core/src/main/scala/org/apache/spark/deploy/mesos/MesosClusterDispatcher.scala

```diff
@@ -50,7 +50,7 @@ private[mesos] class MesosClusterDispatcher(
   extends Logging {
 
   private val publicAddress = Option(conf.getenv("SPARK_PUBLIC_DNS")).getOrElse(args.host)
-  private val recoveryMode = conf.get("spark.mesos.deploy.recoveryMode", "NONE").toUpperCase()
+  private val recoveryMode = conf.get("spark.deploy.recoveryMode", "NONE").toUpperCase()
```

Comment:
I'm still wondering about this as well. I think having some backward compatible makes sense at least for a version. Let me add a warning message when it's set and use the value if it's set as well.